src/screens/OtherProfile/index.js

```diff
@@ -126,6 +121,34 @@ const OtherProfile = () => {
     setOtherId(params?.data?.other_id);
   }, [params.data]);
 
+  const getOtherFeeds = async (offset = 0) => {
+    const otherId = other_id;
+    try {
+      const cacheFeed = StorageUtils.otherProfileFeed.getForKey(otherId);
+      if (cacheFeed) {
+        setOtherProfileFeed(JSON.parse(cacheFeed), dispatchOtherProfile);
+      }
+      const result = await getOtherFeedsInProfile(otherId, offset);
+      const {data: feedOtherProfile} = result;
+      const {mapNewData} = mappingColorFeed({
+        dataFeed: feedOtherProfile,
+        dataCache: cacheFeed
+      });
+      if (offset === 0) {
+        setOtherProfileFeed(mapNewData, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(otherId, JSON.stringify(mapNewData));
+      } else {
+        const clonedFeeds = [...feeds, ...mapNewData];
+        setOtherProfileFeed(clonedFeeds, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(otherId, JSON.stringify(clonedFeeds));
+      }
+      setLoading(false);
+      setPostOffset(Number(result.offset));
+    } catch (e) {
+      setLoading(false);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getOtherFeeds` function has been introduced. It fetches feeds for the other user's profile and stores them in local storage. Ensure that error handling is properly implemented and that data privacy is maintained when storing user data in local storage.
```diff
+ const getOtherFeeds = async (offset = 0) => {
+   const otherId = other_id;
+   try {
+     const cacheFeed = StorageUtils.otherProfileFeed.getForKey(otherId);
+     if (cacheFeed) {
+       setOtherProfileFeed(JSON.parse(cacheFeed), dispatchOtherProfile);
+     }
+     const result = await getOtherFeedsInProfile(otherId, offset);
+     const {data: feedOtherProfile} = result;
+     const {mapNewData} = mappingColorFeed({
+       dataFeed: feedOtherProfile,
+       dataCache: cacheFeed
+     });
+     if (offset === 0) {
+       setOtherProfileFeed(mapNewData, dispatchOtherProfile);
+       StorageUtils.otherProfileFeed.setForKey(otherId, JSON.stringify(mapNewData));
+     } else {
+       const clonedFeeds = [...feeds, ...mapNewData];
+       setOtherProfileFeed(clonedFeeds, dispatchOtherProfile);
+       StorageUtils.otherProfileFeed.setForKey(otherId, JSON.stringify(clonedFeeds));
+     }
+     setLoading(false);
+     setPostOffset(Number(result.offset));
+   } catch (e) {
+     setLoading(false);
+   }
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->